### ISSUE TITLE
Read-Only Properties Cause Exception when Using DI

### ISSUE THREAD

**Issue by rlvandaveer** [natemcmaster/CommandLineUtils#209]
**Describe the bug**
When a read-only property is added to a command, like version, an exception is thrown when UseConstructorInjection is called.

**To Reproduce**
1. Add a read-only Version property to your dependency inject sample.
2. Run the project
3. System.InvalidOperationException: "Could not find a way to set [Insert Type Name Here].Version"

**Expected behavior**
Read-Only properties should be ignored as they are either going to be set by the developer in the constructor or retrieved at runtime via logic.

For now I am going to implement my version as a method to avoid this issue.

**Comment by natemcmaster** [comment]
Can you share sample code?

**Comment by rlvandaveer** [comment]
Sure. How would you like a sample, is an excerpt in a reply sufficient?

**Comment by natemcmaster** [comment]
Yeah, excerpt is fine. You said you used a sample from this repo, so if you can just show me what you modified, that would help a lot.

**Comment by rlvandaveer** [comment]
The difference makes sense. Why would one want to set a read-only property though? It seems to me that a read-only property should be controlled by the type itself. I figured the solution to this would be to remove the InvalidOperationException and return some sort of "NoOp delegate from the `ReflectionHelper` but I didn't look at the downstream impacts of that.
